src/tokens/TokenDetails.tsx

```diff
@@ -76,12 +78,19 @@ export default function TokenDetailsScreen({ route }: Props) {
             {token.name}
           </Text>
         </View>
-        <TokenDisplay
-          amount={token.balance}
-          tokenId={tokenId}
-          style={styles.balance}
-          testID="TokenDetails/Balance"
-        />
+        {token?.priceUsd !== null ? (
+          <TokenDisplay
+            amount={1}
+            tokenId={tokenId}
+            style={styles.assetValue}
+            testID="TokenDetails/AssetValue"
+          />
+        ) : (
+          <Text style={styles.assetValue} testID="TokenDetails/AssetValue">
+            {localCurrencySymbol ?? '$'}
+            {' --'}
+          </Text>
```

Comment:
Looking at token details, looks like we end up showing `-` if either `tokenInfo.priceUsd` is null or if `localCurrencyExchangeRate` is also null. Wonder if we should use the same condition here and for showing "Price unavailable" text.

Reply:
I went a different direction and added an errorFallback prop to TokenDisplay

Reply:
to avoid duplicating logic from TokenDisplay and having to somehow keep it in sync

Reply:
sounds like you might be concerned about a separate corner case, where TokenDisplay is in an error state but PriceInfo is not. I think the main purpose of this PR is to show the correct data, and not to deal with the other bugs in this component related to error states, so I suggest we deal with that separately, or maybe skip it, since a solution is not obvious to me and I'm not sure it will ever come up

Reply:
I guess we'd need to just ditch TokenDisplay and use something else

Reply:
ok now I'm interested in trying that lol